File: 프로그래머스/2/12914. 멀리 뛰기/멀리 뛰기.cpp
#include <string>
#include <vector>

using namespace std;
//해당 코드는 결과론적으로 봤을때 피보나치와 완전 동일 ㅋㅋ 훼이크지렷
long long solution(int n) {
    long long arr[2000];
    arr[0] = 1;
    arr[1] = 1;
    for(int i = 2; i <= n; i++)
        arr[i] = (arr[i - 2] + arr[i - 1]) % 1234567;
    
    return arr[n];
}

File: 프로그래머스/2/12914. 멀리 뛰기/멀리 뛰기.java
class Solution {
    public long solution(int n) {
        int[] ways = new int[n+1];
        ways[0] = 1; // 합이 0인 경우의 수는 1
        
        for (int i = 1; i <= n; i++) {
            ways[i] += ways[i-1] % 1234567;
            if (i >= 2) {
                ways[i] += ways[i-2] % 1234567;
            }
        }
        
        return ways[n] % 1234567;
    }
}